ui/app/components/MethodTreeComponent/MethodTreeComponent.js

```diff
@@ -29,27 +30,36 @@ const dedupeNodes = allNodes => nextNodesAccessorField => (nodes) => {
       globalOnCPUSum += curr.onCPU;
       newCurr.onStack = curr.onCPU;
     }
+    let key;
     const evaluatedOnStack = childOnStack || newCurr.onStack;
     newCurr.onStack = evaluatedOnStack;
     // only do this if it's bottom-up or nextNodesAccessorField === 'parent'
     // change structure of parent array, store onStack also
     if (nextNodesAccessorField === 'parent') {
-      newCurr[nextNodesAccessorField] = newCurr.name
-        ? [[...newCurr[nextNodesAccessorField], evaluatedOnStack]] : [];
+      if(Array.isArray(newCurr.parent) && !(Number.isInteger(newCurr.parent[0]) || Array.isArray(newCurr.parent[0]))) {
+        //Only the topmost node of hotmethod has parent structure as Array(0) of Object
+        key = simpleHash(newCurr.name.toString());  //if top level node in hotmethod view then hash by name not by name+lineno
+      }else{
+        key = simpleHash(newCurr.name.toString() + ':' + newCurr.lineNo.toString());
```

Comment:
hash wont work because of collisions. See if you can directly use the "methodName:LineNo".